Tidy up of CHANNEL resource

nekokatt opened the pull request:
- Removed documentation for the `bulk_delete` endpoint. This has been marked as deprecated
    for like two years now (in favour of `bulk-delete`). No point encouraging users to even consider it 
    exists anymore. If a user has not read these docs for two years, their implementation most likely 
    no longer works anyway...
- Added notes about proxy_url, height, width to embed documentation directly, since it should 
    be kept with the other information about embed constraints for consistency.
- Converted deprecation warning about `type` in embeds to a warning dialog box thing.
- Added missing punctuation to `Group DM` docs.
- Added that the `nick` field in Group DM Add Recipient` is optional.
- Changed wording of `204 empty response` to just say `204 No Content`. This way the
    user is encouraged by RFC-7231 to ignore the response rather than explicitly
    assert that it is empty. This can also reduce some confusion with accidentally
    receiving a Content-Type in a 204 response, as they can just not look for it.
    Also this prevents confusion between an empty object and an empty response.
- Changed wording of stuff `returning a list` to `returning an array`, since it is JSON.
- Changed a warning about multipart fields to an info, it isn't warning anything other than
    to not send random undocumented garbage.
- Fixed formatting inconsistencies (quoting status codes, quoting event names and using their real name, etc).
- Fixed line lengths; most people edit this on GitHub, so it just becomes a mess of altering line lengths otherwise...

Commit:
Tidy up of CHANNEL resource

- Removed documentation for the `bulk_delete` endpoint. This has been marked as deprecated
    for like two years. No point encouraging users to even consider it exists anymore. If a user
    has not read these docs for two years, their implementation most likely no longer works
    anyway...
- Added notes about proxy_url, height, width to embed documentation directly, since it should 
    be kept with the other information about embed constraints for consistency.
- Converted deprecation warning about `type` in embeds to a warning dialog box thing.
- Added missing punctuation to `Group DM` docs.
- Added that the `nick` field in Group DM Add Recipient` is optional.
- Changed wording of `204 empty response` to just say `204 No Content`. This way the
    user is encouraged by RFC-7231 to ignore the response rather than explicitly
    assert that it is empty. This can also reduce some confusion with accidentally
    receiving a Content-Type in a 204 response, as they can just not look for it.
    Also this prevents confusion between an empty object and an empty response.
- Changed wording of stuff `returning a list` to `returning an array`, since it is JSON.
- Changed a warning about multipart fields to an info, it isn't warning anything other than
    to not send random undocumented garbage.
- Fixed formatting inconsistencies (quoting status codes, quoting event names and using their real name, etc).
- Fixed line lengths; most people edit this on GitHub, so it just becomes a mess of altering line lengths otherwise...

advaith1 commented on docs/resources/Channel.md:
```diff
@@ -867,38 +960,38 @@ Create a new [invite](#DOCS_RESOURCES_INVITE/invite-object) object for the chann
-> warn
-> The max pinned messages is 50.
+> info
+> The maximum number of pinned messages per text channel is currently set to 50.
```
news channels too

vladfrangu commented on docs/resources/Channel.md:
```diff
@@ -465,13 +488,16 @@ Embed types are "loosely defined" and, for the most part, are not used by our cl
+| ------- | ------- | ---------------------------------------------------------------------------------------------------------------- |
+| name    | string  | name of the field                                                                                                |
+| value   | string  | value of the field                                                                                                |
+| inline? | boolean | whether or not this field should display inline. The default is to render this inline if not specified. |
```
This suggestion may need some better wording, so please take it with a grain of salt
```suggestion
| inline? | boolean | whether or not this field should display inline. By default, the field will not be rendered inline unless specified. |
```

![image](https://user-images.githubusercontent.com/17960496/82755552-e515d280-9ddc-11ea-98ad-3c6770196a4a.png)

Commit:
Update docs/resources/Channel.md

@vladfrangu's suggestion that the wording on inline defaults needed inversion.

Co-authored-by: Vlad Frangu <kingdgrizzle@gmail.com>

Commit:
Update Channel.md

@advaith1's comment that news channels also have a 50 pin limit.

advaith1 commented on docs/resources/Channel.md:
```diff
@@ -975,7 +975,7 @@ Returns all pinned messages in the channel as an array of [message](#DOCS_RESOUR
 Pin a message in a text channel. Requires the `MANAGE_MESSAGES` permission. Returns a `204 No Content` on success.
```
```suggestion
Pin a message in a text or news channel. Requires the `MANAGE_MESSAGES` permission. Returns a `204 No Content` on success.
```
needs to be updated here too

advaith1 commented on docs/resources/Channel.md:
```diff
@@ -975,7 +975,7 @@ Returns all pinned messages in the channel as an array of [message](#DOCS_RESOUR
 
 > info
-> The maximum number of pinned messages per text channel is currently set to 50.
+> The maximum number of pinned messages per text channel and news channel is currently set to 50.
```
maybe just leave it as "per channel" tbh? saying the isn't really necessary, as there are no different limits for other channel types

```suggestion
> The maximum number of pinned messages per channel is currently set to 50.
```

Commit:
Update docs/resources/Channel.md

Co-authored-by: Advaith <advaithj1@gmail.com>
